test/unit/processing/process-payment-file.test.js

```diff
@@ -0,0 +1,69 @@
+const processPaymentFile = require('../../../app/processing/process-payment-file')
+
+jest.mock('../../../app/processing/batch')
+const batch = require('../../../app/processing/batch')
+
+jest.mock('../../../app/storage')
+const storage = require('../../../app/storage')
+
+jest.mock('../../../app/processing/reprocess-if-needed')
+const reprocessIfNeeded = require('../../../app/processing/reprocess-if-needed')
+
+jest.mock('../../../app/processing/download-and-parse')
+const downloadAndParse = require('../../../app/processing/download-and-parse')
+const { sfiPilot } = require('../../../app/schemes')
+
+global.console.error = jest.fn()
+global.console.log = jest.fn()
+
+let filename
+
+describe('Process payment file', () => {
+  beforeEach(() => {
+    filename = 'SITIELM0001_AP_20220317104956617.dat'
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  test('File has already been processed', async () => {
+    reprocessIfNeeded.mockResolvedValue(true)
+    await processPaymentFile(filename, sfiPilot)
+    expect(batch.nextSequenceId).not.toHaveBeenCalled()
+  })
+
+  test('An error is thrown with a console.error', async () => {
+    reprocessIfNeeded.mockImplementation(() => { throw new Error() })
+    await processPaymentFile(filename, sfiPilot)
+    expect(console.error).toHaveBeenCalledTimes(2)
+  })
```

Comment:
These 2 test names are very poor and don't tell me what we're wanting to test or expecting the outcome to be

Reply:
Also, interestingly we're using 2 different implementations of defining mocked func returns -- is this something we should standardise, pick 1 and use it?

Reply:
Tests implementation is also not adding value.  Have removed.

For standardisation, they're both doing different things.  `.mockResolvedValue` is mocking the end return value of a function.  Whilst `mockImplementation` is mocking the behaviour of the function.  If you want to simulate a function throwing an error, it's not possible to do with `mockResolvedValue`.